PanicKit integration

eighthave opened the pull request:
This adds a settings screen called "Panic Responders" that lets the user configure which apps will receive the PanicKit panic trigger when the panic button is pressed.  This also hooks up the panic button to send the PanicKit trigger message to all enabled apps.

You can see an example of this setup here:
https://www.youtube.com/watch?v=mS1gstS6YS8

It is demoing [Ripple](https://github.com/guardianproject/ripple), [Orweb](https://github.com/guardianproject/orweb), and a little bit of [Zom](https://github.com/zom/zom-android).

jmatsushita commented:
Awesome to see this finally implemented! Thanks a lot for your work that's a great Christmas present :)

I've merge this into a ```panickit``` branch and did some basic testing.  When I click on "EDIT" to configure the Zom responder, there is a screen asking to open with Zom or the Panic Trigger App. I intuitively tried with Zom but that didn't work, and then reverted to Panic Trigger App which seems to be the correct link. Gladly I had asked "Just Once" so I can change this. But this could prove to be a usability problem?

I'll merge #156 onto ```panickit``` now so you can restart from there and keep this PR open if you want to do further work, alternatively feel free to close this one and start another one!

eighthave commented:
That Zom/Panic Trigger App chooser should never show, that was a bug on
my part, I set the Intent Action to null rather than
Panic.ACTION_CONNECT.  I added that fix here.  I rebased my commits on
top on the main ```dev``` branch and force pushed them again here.  So
you can just merge this, and then delete your ```panickit``` branch.

jmatsushita commented:
Hey, if you could make the base of this PR the panickit branch then I could just accept the merge as I'm keeping dev clean for our upcoming 1.5.0 release. Thanks!

eighthave commented:
My rebased version updated the stuff for the latest work in dev, so
perhaps that's the best place to test from without merging?  You can
test my branch without merging by doing:

 git remote add eighthave https://github.com/eighthave/PanicButton
 git fetch eighthave
 git checkout eighthave/panickit-integration

Then if that works for you, you can update your panickit branch by doing:

 git branch -D panickit  # deletes the local copy of the branch
 git branch panickit eighthave/panickit-integration # make a new
panickit branch

Then to force push your local panickit, overwriting the old one:

 git push -f origin panickit

I tried rebasing my panickit-integration branch on your panickit branch,
but it already includes the new stuff from dev, so its not easy to do.

kevinmday commented:
Hi all. I'm new and just opened a new repository called panicbutton-ccw. like panicbutton.io, my idea is for a panicbutton specifically for CCW holders and other 1st responders. citizen defenders would receive any triggered alert originating within their own user-defined response area. this is a public safety initiative aimed at better protecting innocent people against terrorism, anarchy, and other mass shooters.

eighthave commented:
Good to see more ideas around panic responses!  PanicKit is meant to make this all possible.  Let me know if you have PanicKit questions.

kevinmday commented:
Thanks! I'll check panickit out on the hub. Kevin M. Day 

```
  From: Hans-Christoph Steiner <notifications@github.com>
```

 To: PanicInitiative/PanicButton PanicButton@noreply.github.com 
Cc: bendor kevinmday@yahoo.com
 Sent: Monday, February 29, 2016 12:39 PM
 Subject: Re: [PanicButton] PanicKit integration (#157)

Good to see more ideas around panic responses! PanicKit is meant to make this all possible. Let me know if you have PanicKit questions.—
Reply to this email directly or view it on GitHub.

eighthave commented:
I rebased this to be on top of the latest _dev_ branch.

eighthave commented:
any word on getting this included?  I'm happy to fix the merge conflicts when you're ready to merge this.

kevinmday commented:
Hello,
I must admit, this was just an idea I had a while back. I don't know how to merge this, but definitely support those who do. 
What do I need to do in order to move this along? Thanks in advance. Kevin 
      From: Hans-Christoph Steiner notifications@github.com
 To: PanicInitiative/PanicButton PanicButton@noreply.github.com 
Cc: bendor kevinmday@yahoo.com; Comment comment@noreply.github.com
 Sent: Friday, June 24, 2016 1:48 AM
 Subject: Re: [PanicInitiative/PanicButton] PanicKit integration (#157)

any word on getting this included? I'm happy to fix the merge conflicts when you're ready to merge this.—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.

eighthave commented:
Ok, I rebased and fixed this to be on top of the latest **dev** branch.  Its ready to be merged! :-)

eighthave commented:
I think its important to the Panic Initiative to see this feature put out in a full release of PanicButton so we have two clear panic button apps available.  How about I make a version of this that is on top of the last official release, so that this would be the only change?  Then we can get that out sooner since the changes would be small and very low risk. We could include some translation updates while we are at it, since I believe Localization Lab recently worked on this app.

Then the code can be integrated in the dev branch and included once that goes through the final review process.
